Do not insert observers for empty sequential modules

raghuramank100 opened the pull request:
Fixes #28375 

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28384 Do not insert observers for empty sequential modules**

Differential Revision: [D18047293](https://our.internmc.facebook.com/intern/diff/D18047293/)

raghuramank10000 committed:
Do not insert observers for empty sequential modules



Differential Revision: [D18047293](https://our.internmc.facebook.com/intern/diff/D18047293/)

[ghstack-poisoned]

hx89 commented on torch/quantization/quantize.py:
```diff
@@ -95,10 +95,10 @@ def add_observer_(module):
+       len(module._modules) == 0 and not isinstance(module, torch.nn.Sequential):
+            # observer and hook will be gone after we swap the module
+            module.add_module('activation_post_process', module.qconfig.activation())
+            module.register_forward_hook(_observer_forward_hook)
```
The indentation might be off?

raghuramank100 replied:
Thanks, fixed now

raghuramank10000 committed:
Update on "Do not insert observers for empty sequential modules"


Differential Revision: [D18047293](https://our.internmc.facebook.com/intern/diff/D18047293/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Do not insert observers for empty sequential modules"


Differential Revision: [D18047293](https://our.internmc.facebook.com/intern/diff/D18047293/)

[ghstack-poisoned]
